torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1724,19 +1726,58 @@ struct to_ir {
   // raise a
   //
   // We ignore the expression following raise
-  void emitRaise(const SourceRange& loc) {
-    const std::string exception = "Exception";
-    auto string_input = insertConstant(*graph, exception, loc);
-    graph->insert(prim::RaiseException, {string_input}, {}, loc);
+  void emitRaise(const Raise& raise) {
+    auto sv = emitSugaredExpr(raise.expr(), 1);
+    Value* value = nullptr;
+    auto exception_sv = std::dynamic_pointer_cast<ExceptionMessageValue>(sv);
+    if (exception_sv == nullptr) {
```

Comment:
The nesting here is a little hard to follow, and with the `if (value == nullptr)` and the `exception_sv` below. Could you try to rewrite / refactor ? maybe by consolidating ExceptionValue & ExceptionMessageValue or having one inherit from the other